Octokit/Models/Response/StringEnum.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    [SuppressMessage("Microsoft.Naming", "CA1711:IdentifiersShouldNotHaveIncorrectSuffix")]
+    public struct StringEnum<TEnum> : IEquatable<StringEnum<TEnum>>
+        where TEnum : struct
+    {
+        private readonly string _stringValue;
+
+        private TEnum? _parsedValue;
+
+        public StringEnum(TEnum parsedValue)
+        {
+            if (!Enum.IsDefined(typeof(TEnum), parsedValue))
+            {
+                throw GetArgumentException(parsedValue.ToString());
+            }
+
+            // Use the SimpleJsonSerializer to serialize the TEnum into the correct string according to the GitHub Api strategy
+            _stringValue = new SimpleJsonSerializer().SerializeEnum(parsedValue as Enum);
```

Comment:
We *really* need to get rid of these before this PR is merged. Allocating a `SimpleJsonSerializer` every time a `StringEnum` is created and parsed is hella expensive when it comes to GC. I tried making it static, but for some reason tests started failing, so it seems it actually has shared mutable state 😢